modules/Profile/components/TeacherProfile/TeacherProfile.tsx

```diff
@@ -0,0 +1,81 @@
+import { Grid, Table, Title, Box } from "@mantine/core";
+import { notifications } from "@mantine/notifications";
+
+import { ITeacherProfileProps } from "./TeacherProfile.interface";
+
+const TeacherProfile: React.FC<ITeacherProfileProps> = ({ user }) => {
+  if (!user || !user.teacher) {
+    notifications.show({
+      title: "Profile Data Unavailable",
+      message: "No teacher profile data is available at the moment.",
+      color: "red",
+    });
+  }
+
+  if (!user || !user.teacher) return null;
+
+  const subjectToTeachRows = user.teacher.subjectsToTeach.map((subject, index) => (
+    <tr key={subject}>
+      <td>
+        <Title order={4} ta="center" c="#4CAF50">
+          {index + 1}
+        </Title>
+      </td>
+      <td>
+        <Title order={4} ta="center" c="#4CAF50">
+          {subject}
+        </Title>
+      </td>
+    </tr>
+  ));
+
+  const profileItems = [
+    { label: "Email", value: user.email },
+    { label: "Gender", value: user.gender || "Not specified" },
+    { label: "First name", value: user.firstName },
+    { label: "Last name", value: user.lastName },
+    { label: "Major Subject", value: user.teacher.majorSubject },
+    { label: "Highest education level", value: user.teacher.highestEducationLevel },
+  ];
```

Comment:
This looks like a helper that takes in user as param. Should be inside the .helpers.ts file